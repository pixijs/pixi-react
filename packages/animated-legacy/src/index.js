export * from '@pixi/react-pixi-animated/src/index';
