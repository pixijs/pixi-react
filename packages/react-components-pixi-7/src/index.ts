/// <reference path="./global.d.ts" />
import type { ComponentsType, PixiComponentType } from '@pixi/react-types';

import {
    AnimatedSprite as AnimatedSpriteComponent,
    BitmapText as BitmapTextComponent,
    Container as ContainerComponent,
    Graphics as GraphicsComponent,
    NineSlicePlane as NineSlicePlaneComponent,
    ParticleContainer as ParticleContainerComponent,
    SimpleMesh as SimpleMeshComponent,
    SimpleRope as SimpleRopeComponent,
    Sprite as SpriteComponent,
    Text as TextComponent,
    TilingSprite as TilingSpriteComponent,
} from './components';
import { applyDefaultProps } from './utils/props';

import type {
    PixiReactContainer,
    ReactStageComponent,
    ReactAnimatedSpriteComponent,
    ReactBitmapTextComponent,
    ReactContainerComponent,
    ReactGraphicsComponent,
    ReactNineSlicePlaneComponent,
    ReactParticleContainerComponent,
    ReactSimpleMeshComponent,
    ReactSimpleRopeComponent,
    ReactSpriteComponent,
    ReactTextComponent,
    ReactTilingSpriteComponent,
} from './types';

/**
 * -------------------------------------------
 * Public API
 * -------------------------------------------
 */

export { Context as AppContext, AppProvider, AppConsumer, withPixiApp } from './stage/provider';
export { useTick, useApp } from './hooks';
export { withFilters } from './hoc';
export { eventHandlers } from './utils/pixi';
export { configurePixiReactStage } from './stage';
export { configurePixiReactRenderAPI } from './render';
export { applyDefaultProps };
export type { PixiReactContainer, ReactStageComponent };

/**
 * Available tag types
 *
 * @type {Object}
 */
export const TYPES = {
    BitmapText: 'BitmapText',
    Container: 'Container',
    Graphics: 'Graphics',
    NineSlicePlane: 'NineSlicePlane',
    ParticleContainer: 'ParticleContainer',
    Sprite: 'Sprite',
    AnimatedSprite: 'AnimatedSprite',
    Text: 'Text',
    TilingSprite: 'TilingSprite',
    SimpleMesh: 'SimpleMesh',
    SimpleRope: 'SimpleRope',
};

export const BitmapText = TYPES.BitmapText as unknown as ReactBitmapTextComponent;
export const Container = TYPES.Container as unknown as ReactContainerComponent;
export const Graphics = TYPES.Graphics as unknown as ReactGraphicsComponent;
export const NineSlicePlane = TYPES.NineSlicePlane as unknown as ReactNineSlicePlaneComponent;
export const ParticleContainer = TYPES.ParticleContainer as unknown as ReactParticleContainerComponent;
export const Sprite = TYPES.Sprite as unknown as ReactSpriteComponent;
export const AnimatedSprite = TYPES.AnimatedSprite as unknown as ReactAnimatedSpriteComponent;
export const Text = TYPES.Text as unknown as ReactTextComponent;
export const TilingSprite = TYPES.TilingSprite as unknown as ReactTilingSpriteComponent;
export const SimpleMesh = TYPES.SimpleMesh as unknown as ReactSimpleMeshComponent;
export const SimpleRope = TYPES.SimpleRope as unknown as ReactSimpleRopeComponent;

const components: ComponentsType = {
    AnimatedSprite: AnimatedSpriteComponent,
    BitmapText: BitmapTextComponent,
    Container: ContainerComponent,
    Graphics: GraphicsComponent,
    NineSlicePlane: NineSlicePlaneComponent,
    ParticleContainer: ParticleContainerComponent,
    SimpleMesh: SimpleMeshComponent,
    SimpleRope: SimpleRopeComponent,
    Sprite: SpriteComponent,
    Text: TextComponent,
    TilingSprite: TilingSpriteComponent,
};

export function configurePixiReactComponents(
    PixiComponent: PixiComponentType,
)
{
    // Register default components in `react-modular`'s COMPONENTS hash
    Object.keys(TYPES).forEach((type) => PixiComponent(type, components[type]));

    return {
        TYPES,
        applyDefaultProps,
    };
}
