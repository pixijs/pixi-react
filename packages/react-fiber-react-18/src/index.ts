import Reconciler from 'react-reconciler';
import type { applyPropsType, ComponentType, PropsType, PixiReactMinimalExpandoContainer } from '@pixi/react-types';
import data from './data.json';
import { makeHostConfig } from './hostconfig';
import type { diffPropertiesType, PixiReactHostConfig, PixiReactReconciler } from './types';

export type { PixiReactHostConfig, PixiReactReconciler };

export function configurePixiReactHostConfig<
    Container extends PixiReactMinimalExpandoContainer,
    Instance extends PixiReactMinimalExpandoContainer,
>({
    COMPONENTS,
    applyDefaultProps,
    diffProperties,
}: {
    COMPONENTS: Record<string, ComponentType<PropsType, Container, Instance>>;
    applyDefaultProps: applyPropsType<PropsType, Instance>;
    diffProperties?: diffPropertiesType<Instance>;
})
{
    return makeHostConfig<Container, Instance>({ COMPONENTS, applyDefaultProps, diffProperties });
}

export function configurePixiReactFiber<
    PixiContainer extends PixiReactMinimalExpandoContainer,
    Instance extends PixiReactMinimalExpandoContainer,
>(hostConfig: PixiReactHostConfig<PixiContainer, Instance>): PixiReactReconciler<PixiContainer, Instance>
{
    const PixiReactFiber = Reconciler(hostConfig);

    PixiReactFiber.injectIntoDevTools({
        bundleType: process.env.NODE_ENV !== 'production' ? 1 : 0,
        version: data.REACT_DOM_VERSION,
        rendererPackageName: data.PACKAGE_NAME,
    });

    return PixiReactFiber;
}
