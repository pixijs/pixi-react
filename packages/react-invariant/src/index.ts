// borrowed from fbjs
export function invariant(condition: boolean, format: string, ...args: string[])
{
    if (process.env.NODE_ENV === 'production')
    {
        return;
    }

    if (!condition)
    {
        let error;

        if (format === undefined)
        {
            error = new Error(
                'Minified exception occurred; use the non-minified dev environment '
                + 'for the full error message and additional helpful warnings.'
            );
        }
        else
        {
            let argIndex = 0;

            error = new Error(format.replace(/%s/g, () => String(args[argIndex++])));
            error.name = 'Invariant Violation';
        }

        // @ts-ignore: as per original fbjs implementation
        error.framesToPop = 1; // Skip invariant's own stack frame.
        throw error;
    }
}
