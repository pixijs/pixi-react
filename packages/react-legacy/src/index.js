export * from '@pixi/react/src/index';
