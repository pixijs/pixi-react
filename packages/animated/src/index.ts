import { TYPES } from '@pixi/react';

import { animated } from './react-spring';

export const BitmapText = animated[TYPES.BitmapText];
export const Container = animated[TYPES.Container];
export const Graphics = animated[TYPES.Graphics];
export const NineSlicePlane = animated[TYPES.NineSlicePlane];
export const ParticleContainer = animated[TYPES.ParticleContainer];
export const Sprite = animated[TYPES.Sprite];
export const AnimatedSprite = animated[TYPES.AnimatedSprite];
export const Text = animated[TYPES.Text];
export const TilingSprite = animated[TYPES.TilingSprite];
export const SimpleMesh = animated[TYPES.SimpleMesh];
export const SimpleRope = animated[TYPES.SimpleRope];
