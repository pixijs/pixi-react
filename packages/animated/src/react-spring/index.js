import { TYPES, applyDefaultProps } from '@pixi/react-pixi';
import { createHost } from '@react-spring/animated';
const primitives = Object.keys(TYPES);

const host = createHost(primitives, {
    applyAnimatedValues(instance, props)
    {
        if (!(instance.nodeType || instance.pluginName))
        {
            return false;
        }
        const applyProps = typeof instance?.applyProps === 'function' ? instance.applyProps : applyDefaultProps;

        applyProps(instance, {}, props);
    },
});

const animated = host.animated;

export { animated };
