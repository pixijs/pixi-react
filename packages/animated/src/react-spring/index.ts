import { applyDefaultProps, TYPES } from '@pixi/react';
import { createHost } from '@react-spring/animated';
const primitives = Object.keys(TYPES);

const host = createHost(primitives, {
    // eslint-disable-next-line consistent-return
    applyAnimatedValues(instance, props)
    {
        if (!(instance.nodeType || instance.pluginName))
        {
            return false;
        }
        const applyProps = typeof instance?.applyProps === 'function' ? instance.applyProps : applyDefaultProps;

        applyProps(instance, {}, props);
    },
});

const animated = host.animated;

export { animated };
