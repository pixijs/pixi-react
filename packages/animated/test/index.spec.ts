import * as index from '../src';

describe('index', () =>
{
    test('export modules for react-animated', () =>
    {
        expect(index).toMatchSnapshot();
    });
});
