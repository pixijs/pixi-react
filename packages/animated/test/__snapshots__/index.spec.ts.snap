// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index export modules for react-animated 1`] = `
{
  "AnimatedSprite": {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
  },
  "BitmapText": {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
  },
  "Container": {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
  },
  "Graphics": {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
  },
  "NineSlicePlane": {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
  },
  "ParticleContainer": {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
  },
  "SimpleMesh": {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
  },
  "SimpleRope": {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
  },
  "Sprite": {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
  },
  "Text": {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
  },
  "TilingSprite": {
    "$$typeof": Symbol(react.forward_ref),
    "render": [Function],
  },
}
`;
